presto-tests/src/main/java/com/facebook/presto/tests/H2QueryRunner.java

```diff
@@ -363,7 +365,16 @@ else if (type instanceof ArrayType) {
                             row.add(null);
                         }
                         else {
-                            row.add(newArrayList((Object[]) array.getArray()));
+                            Object[] elements = (Object[]) array.getArray();
+                            Type elementType = ((ArrayType) type).getElementType();
+                            if (elementType instanceof ArrayType) {
+                                row.add(Arrays.stream(elements)
+                                        .map(v -> v == null ? null : newArrayList((Object[]) v))
+                                        .collect(Collectors.toList()));
```

Comment:
Is there a reason it must be mutable?

Reply:
@tdcmeehan ImmutableList doesn't allow null elements, but this list may have nulls.